client.go

```diff
@@ -182,6 +224,19 @@ func (c *Client) Do(req *Request) (*http.Response, error) {
 		req.Method, req.URL, c.RetryMax+1)
 }
 
+// Try to read the response body so we can reuse this connection.
+func (c *Client) drainBody(body io.ReadCloser) {
+	defer body.Close()
```

Comment:
I am a bit confused as to whether the `CheckRetry` is supposed to close the response body or not. Can you clarify?